### ISSUE TITLE
Javascript code in script-tag not executing in React

### ISSUE THREAD

**Issue by meesfrenkelfrank** [remarkablemark/html-react-parser#482]
I am getting this example HTML markup back from api:

`const html = "<div class='my-class' data-src='/event/889979743'></div> <script>alert('Hello world')</script>";`

Then I parse this HTML in my React component like:
`return <div data-testid="html-paragraph">{parser(html)}</div>;`

Why is the parsed Javascript code  `<script>alert('Hello world')</script>`  not executing?

Here my [code sandbox](https://codesandbox.io/s/html-react-parser-v801ww?file=/src/Widget.tsx).

## Reproducible Demo
https://codesandbox.io/s/html-react-parser-v801ww?file=/src/Widget.tsx

**Comment by remarkablemark** [comment]
This is expected behavior. See [Are <script> tags parsed?](https://github.com/remarkablemark/html-react-parser/tree/v1.4.8#are-script-tags-parsed)

**Comment by meesfrenkelfrank** [comment]
@remarkablemark ok thanks. Not in above example but the real script getting back from api (and parsed through `html-react-parser`),  how to handle this properly, so below script gets executed?
```
<div class="my-class" data-src="/event/66478667"></div>
<script>(function(d, s, id) {var js,ijs=d.getElementsByTagName(s)[0];if(d.getElementById(id))return;js=d.createElement(s);js.id=id;js.src="https://embed.widget.js";ijs.parentNode.insertBefore(js, ijs);}(document, 'script', 'my-js'));</script>
```

**Comment by remarkablemark** [comment]
@meesfrenkelfrank you can use [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) (although I don't recommend it because it's [unsafe](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!))

Here's an example of how to do it with [replace](https://github.com/remarkablemark/html-react-parser/tree/v1.4.8#replace):

```js
parse(<script>alert("test");</script>, {
  replace(domNode) {
    if (domNode.type === 'script') {
      eval(domNode.children[0].data);
    }
  },
});
```

See [Replit](https://replit.com/@remarkablemark/html-react-parser-482#script.js)

**Comment by meesfrenkelfrank** [comment]
Ok thanks. But I get Property `'children' does not exist on type 'DOMNode'.`? When I try:
```
if (domNode instanceof Element && domNode.type === 'script') {
 console.log('Children', domNode.children);
}
```
It's not returning anything. TypeScript is 'inferring `Element` from general base class from which all objects in a Document inherit.

**Comment by remarkablemark** [comment]
@meesfrenkelfrank has your issue been resolved in #591? If so, can this issue be closed?
